### PR TITLE
update documentation for Threshold

### PR DESCRIPTION
**Pull request by karhohs**

### CODE REVIEW

File: cellprofiler/modules/threshold.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 **Apply Threshold** sets pixel intensities below or above a certain
```
Comment (mcquin) [inline cellprofiler/modules/threshold.py]: `**Threshold**`

Comment (mcquin) [reply cellprofiler/modules/threshold.py]: Also, the module title in the line above should be changed to `Threshold`

File: cellprofiler/modules/threshold.py [hunk 2]
```diff
@@ -106,99 +102,87 @@ def create_settings(self):
...
-            <p><b>References</b></p>
-            <ul>
-                <li>Sezgin M, Sankur B (2004) "Survey over image thresholding techniques and quantitative
-                performance evaluation." <i>Journal of Electronic Imaging</i>, 13(1), 146-165. (<a href=
-                "http://dx.doi.org/10.1117/1.1631315">link</a>)
-                </li>
-            </ul>
-            <p></p>
+            doc="""\
+*(Used only if {TS_GLOBAL} is selected for thresholding scope)*
```
Comment (mcquin) [inline cellprofiler/modules/threshold.py]: `"{TS_GLOBAL}"`

File: cellprofiler/modules/threshold.py [hunk 3]
```diff
@@ -106,99 +102,87 @@ def create_settings(self):
...
+|image0|  An automatically-calculated threshold adapts to changes in
+lighting/staining conditions between images and is usually more
+robust/accurate. In the vast majority of cases, an automatic method is
+sufficient to achieve the desired thresholding, once the proper method
+is selected. In contrast, an advantage of a manually-entered number is
+that it treats every image identically, so use this option when you have
+a good sense for what the threshold should be across all images. To help
+determine the choice of threshold manually, you can inspect the pixel
+intensities in an image of your choice. {HELP_ON_PIXEL_INTENSITIES}.
+|image1|  The manual method is not robust with regard to slight changes
```
Comment (mcquin) [inline cellprofiler/modules/threshold.py]: Add a newline above this line.

File: cellprofiler/modules/threshold.py [hunk 4]
```diff
@@ -106,99 +102,87 @@ def create_settings(self):
...
+
+|image0|  An automatically-calculated threshold adapts to changes in
+lighting/staining conditions between images and is usually more
+robust/accurate. In the vast majority of cases, an automatic method is
+sufficient to achieve the desired thresholding, once the proper method
+is selected. In contrast, an advantage of a manually-entered number is
+that it treats every image identically, so use this option when you have
+a good sense for what the threshold should be across all images. To help
+determine the choice of threshold manually, you can inspect the pixel
+intensities in an image of your choice. {HELP_ON_PIXEL_INTENSITIES}.
```
Comment (mcquin) [inline cellprofiler/modules/threshold.py]: Remove the `.` at the end of this line. `HELP_ON_PIXEL_INTENSITIES` ends with a `.`

File: cellprofiler/modules/threshold.py [hunk 5]
```diff
@@ -106,99 +102,87 @@ def create_settings(self):
...
+   negligible background, or if the input image is the probability map
+   output of the **ClassifyPixels** module (in which case, a value of
+   0.5 should be chosen). If the input image is already binary (i.e.,
+   where the foreground is 1 and the background is 0), a manual value of
+   0.5 will identify the objects.
+-  *{TM_MEASUREMENT}:* Use a prior image measurement as the threshold.
+   The measurement should have values between zero and one. This
+   strategy can be used to apply a pre-calculated threshold imported as
+   per-image
+-  Sezgin M, Sankur B (2004) “Survey over image thresholding techniques
```
Comment (mcquin) [inline cellprofiler/modules/threshold.py]: Add this above this line:

```python
**References**

- Sezgin M, ...
```

File: cellprofiler/modules/threshold.py [hunk 6]
```diff
@@ -219,61 +203,68 @@ def create_settings(self):
...
-            <p><b>References</b></p>
-            <ul>
-                <li>Sezgin M, Sankur B (2004) "Survey over image thresholding techniques and quantitative
-                performance evaluation." <i>Journal of Electronic Imaging</i>, 13(1), 146-165. (<a href=
-                "http://dx.doi.org/10.1117/1.1631315">link</a>)
-                </li>
-            </ul>
-            <p></p>
+            doc="""\
+*(Used only if {TS_ADAPTIVE} is selected for thresholding scope)*
```
Comment (mcquin) [inline cellprofiler/modules/threshold.py]: `"{TS_ADAPTIVE}"`

File: cellprofiler/modules/threshold.py [hunk 7]
```diff
@@ -219,61 +203,68 @@ def create_settings(self):
...
+
+|image0|  An automatically-calculated threshold adapts to changes in
+lighting/staining conditions between images and is usually more
+robust/accurate. In the vast majority of cases, an automatic method is
+sufficient to achieve the desired thresholding, once the proper method
+is selected. In contrast, an advantage of a manually-entered number is
+that it treats every image identically, so use this option when you have
+a good sense for what the threshold should be across all images. To help
+determine the choice of threshold manually, you can inspect the pixel
+intensities in an image of your choice. {HELP_ON_PIXEL_INTENSITIES}.
```
Comment (mcquin) [inline cellprofiler/modules/threshold.py]: `{HELP_ON_PIXEL_INTENSITIES}.` -> `{HELP_ON_PIXEL_INTENSITIES}`

File: cellprofiler/modules/threshold.py [hunk 8]
```diff
@@ -219,61 +203,68 @@ def create_settings(self):
...
+|image1|  The manual method is not robust with regard to slight changes
```
Comment (mcquin) [inline cellprofiler/modules/threshold.py]: Add a new line before this line.

File: cellprofiler/modules/threshold.py [hunk 9]
```diff
@@ -417,9 +418,11 @@ def create_settings(self):
             0.05,
             minval=0,
             maxval=1,
-            doc="""
-            <i>(Used only when customizing the {TM_ROBUST_BACKGROUND} method)</i><br>
-            Discard this fraction of the pixels in the image starting with those of the lowest intensity.
+            doc="""\
+*(Used only when customizing the {TM_ROBUST_BACKGROUND} method)*
```
Comment (mcquin) [inline cellprofiler/modules/threshold.py]: `"{TM_ROBUST_BACKGROUND}"`

File: cellprofiler/modules/threshold.py [hunk 10]
```diff
@@ -430,9 +433,10 @@ def create_settings(self):
             0.05,
             minval=0,
             maxval=1,
-            doc="""
-            <i>(Used only when customizing the {TM_ROBUST_BACKGROUND} method)</i><br>
-            Discard this fraction of the pixels in the image starting with those of the highest intensity.
+            doc="""\
+*(Used only when customizing the {TM_ROBUST_BACKGROUND} method)*
```
Comment (mcquin) [inline cellprofiler/modules/threshold.py]: `"{TM_ROBUST_BACKGROUND}"`

Comment (mcquin) [inline cellprofiler/modules/threshold.py]: Add a new line after the `*(Used only if...)*` line.

File: cellprofiler/modules/threshold.py [hunk 11]
```diff
@@ -441,19 +445,22 @@ def create_settings(self):
...
-                is a good choice if the cell density is variable or high.</li>
-                <li><i>{RB_MEDIAN}</i>: Use the median of the pixels. This is a good choice if, for all images,
-                more than half of the pixels are in the background after removing outliers.</li>
-                <li><i>{RB_MODE}</i>: Use the most frequently occurring value from among the pixel values. The
-                {TM_ROBUST_BACKGROUND} method groups the intensities into bins (the number of bins is the
-                square root of the number of pixels in the unmasked portion of the image) and chooses the
-                intensity associated with the bin with the most pixels.</li>
-            </ul>
+            doc="""\
+*(Used only when customizing the {TM_ROBUST_BACKGROUND} method)*
```
Comment (mcquin) [inline cellprofiler/modules/threshold.py]: `"{TM_ROBUST_BACKGROUND}"`

File: cellprofiler/modules/threshold.py [hunk 12]
```diff
@@ -465,13 +472,13 @@ def create_settings(self):
...
             [RB_SD, RB_MAD],
-            doc="""
-            <i>(Used only when customizing the {TM_ROBUST_BACKGROUND} method)</i><br>
-            Robust background adds a number of deviations (standard or MAD) to the average to get the final
-            background. This setting chooses the method used to assess the variance in the pixels, after
-            removing outliers.<br>
-            Choose one of <i>{RB_SD}</i> or <i>{RB_MAD}</i> (the median of the absolute difference of the pixel
-            intensities from their median).
+            doc="""\
+*(Used only when customizing the {TM_ROBUST_BACKGROUND} method)*
```
Comment (mcquin) [inline cellprofiler/modules/threshold.py]: `"{TM_ROBUST_BACKGROUND}"`

Comment (mcquin) [inline cellprofiler/modules/threshold.py]: Add a new line after this line.

File: cellprofiler/modules/threshold.py [hunk 13]
```diff
@@ -482,14 +489,17 @@ def create_settings(self):
...
-            doc="""
-            <i>(Used only when customizing the {TM_ROBUST_BACKGROUND} method)</i><br>
-            Robust background calculates the variance, multiplies it by the value given by this setting and
-            adds it to the average. Adding several deviations raises the background value above the average,
-            which should be close to the average background, excluding most background pixels. Use a larger
-            number to exclude more background pixels. Use a smaller number to include more low-intensity
-            foreground pixels. It's possible to use a negative number to lower the threshold if the averaging
-            method picks a threshold that is within the range of foreground pixel intensities.
+            doc="""\
+*(Used only when customizing the {TM_ROBUST_BACKGROUND} method)*
```
Comment (mcquin) [inline cellprofiler/modules/threshold.py]: `"{TM_ROBUST_BACKGROUND}"`

Comment (mcquin) [inline cellprofiler/modules/threshold.py]: Add a new line after this line.

File: cellprofiler/modules/threshold.py (hunk 8 above)
Comment (bethac07) [inline cellprofiler/modules/threshold.py]: The settings block after the image should be indented.

File: cellprofiler/modules/threshold.py [hunk 14]
```diff
@@ -63,33 +63,29 @@ def create_settings(self):
...
+calculate the threshold. These options allow you to calculate a
+threshold based on the whole image or based on image sub-regions.
+  
+The choices for the threshold strategy are:
+
+-  *{TS_GLOBAL}:* Calculate a single threshold value based on the
+   unmasked pixels of the input image and use that value to classify
+   pixels above the threshold as foreground and below as background.
+   
+   |image0|  This strategy is fast and robust, especially if the
```
Comment (bethac07) [inline cellprofiler/modules/threshold.py]: The settings block after the image should be indented.

File: cellprofiler/modules/threshold.py [hunk 15]
```diff
@@ -106,99 +102,87 @@ def create_settings(self):
...
+will be considered foreground (region(s) of interest) or background. A
+higher threshold value will result in only the brightest regions being
+identified, whereas a lower threshold value will include dim regions.
+You can have the threshold automatically calculated from a choice of
+several methods, or you can enter a number manually between 0 and 1
+for the threshold.
+
+Both the automatic and manual options have advantages and disadvantages.
+
+|image0|  An automatically-calculated threshold adapts to changes in
```
Comment (bethac07) [inline cellprofiler/modules/threshold.py]: The settings block after the image should be indented.

File: cellprofiler/modules/threshold.py (hunk 3 above)
Comment (bethac07) [inline cellprofiler/modules/threshold.py]: The settings block after the image should be indented.

File: cellprofiler/modules/threshold.py [hunk 16]
```diff
@@ -219,61 +203,68 @@ def create_settings(self):
...
+your images, you might check whether the automatically calculated
+threshold was unusually high or low compared to the other images. See
+the **FlagImage** module if you would like to flag an image based on the
+threshold value.
+
+There are a number of methods for finding thresholds automatically:
+
+-  *{TM_OTSU}:* This approach calculates the threshold separating the
+   two classes of pixels (foreground and background) by minimizing the
+   variance within the each class. |image2|  This method is a good
```
Comment (bethac07) [inline cellprofiler/modules/threshold.py]: The settings block after the image should be indented.

File: cellprofiler/modules/threshold.py [hunk 17]
```diff
@@ -219,61 +203,68 @@ def create_settings(self):
...
+   initial approach if you do not know much about the image
+   characteristics of all the images in your experiment, especially if
+   the percentage of the image covered by foreground varies
+   substantially from image to image. |image3|  Our implementation of
```
Comment (bethac07) [inline cellprofiler/modules/threshold.py]: The settings block after the image should be indented.

File: cellprofiler/modules/threshold.py [hunk 18]
```diff
@@ -63,33 +63,29 @@ def create_settings(self):
...
+   unmasked pixels of the input image and use that value to classify
+   pixels above the threshold as foreground and below as background.
+   
+   |image0|  This strategy is fast and robust, especially if the
+   background is uniformly illuminated.
+-  *{TS_ADAPTIVE}:* Partition the input image into tiles and calculate
+   thresholds for each tile. For each tile, the calculated threshold is
+   applied only to the pixels within that tile.
+   
+   |image1|  This method is slower but can produce better results for
```
Comment (bethac07) [inline cellprofiler/modules/threshold.py]: The settings block after the image should be indented.

File: cellprofiler/modules/threshold.py [hunk 19]
```diff
@@ -106,99 +102,87 @@ def create_settings(self):
...
+your images, you might check whether the automatically calculated
+threshold was unusually high or low compared to the other images. See
+the **FlagImage** module if you would like to flag an image based on the
+threshold value.
+
+There are a number of methods for finding thresholds automatically:
+            
+-  *{TM_OTSU}:* This approach calculates the threshold separating the
+   two classes of pixels (foreground and background) by minimizing the
+   variance within the each class. |image2|  This method is a good
```
Comment (bethac07) [inline cellprofiler/modules/threshold.py]: The settings block after the image should be indented.

File: cellprofiler/modules/threshold.py [hunk 20]
```diff
@@ -106,99 +102,87 @@ def create_settings(self):
...
+   initial approach if you do not know much about the image
+   characteristics of all the images in your experiment, especially if
+   the percentage of the image covered by foreground varies
+   substantially from image to image. |image3|  Our implementation of
```
Comment (bethac07) [inline cellprofiler/modules/threshold.py]: The settings block after the image should be indented.

File: cellprofiler/modules/threshold.py [hunk 21]
```diff
@@ -106,99 +102,87 @@ def create_settings(self):
...
+   Otsu’s method allows for assigning the threshold value based on
+   splitting the image into either two classes (foreground and
+   background) or three classes (foreground, mid-level, and background).
+   See the help below for more details.
+-  *{TM_ROBUST_BACKGROUND}:* This method assumes that the background
+   distribution approximates a Gaussian by trimming the brightest and
+   dimmest 5% of pixel intensities. It then calculates the mean and
+   standard deviation of the remaining pixels and calculates the
+   threshold as the mean + 2 times the standard deviation. |image4|
```
Comment (bethac07) [inline cellprofiler/modules/threshold.py]: The settings block after the image should be indented.

File: cellprofiler/modules/threshold.py [hunk 22]
```diff
@@ -106,99 +102,87 @@ def create_settings(self):
...
+   standard deviation of the remaining pixels and calculates the
+   threshold as the mean + 2 times the standard deviation. |image4| 
+   This thresholding method can be helpful if the majority of the image
+   is background. It can also be helpful if your images vary in overall
+   brightness, but the objects of interest are consistently *N* times
+   brighter than the background level of the image.
+-  *{TM_LI}:*
+-  *{TM_MANUAL}:* Enter a single value between zero and one that
+   applies to all cycles and is independent of the input image.
+   |image5|  This approach is useful if the input image has a stable or
```
Comment (bethac07) [inline cellprofiler/modules/threshold.py]: The settings block after the image should be indented.

File: cellprofiler/modules/threshold.py [hunk 23]
```diff
@@ -352,48 +368,55 @@ def create_settings(self):
             value=0.0,
             minval=0.0,
             maxval=1.0,
-            doc="""
-            <i>(Used only if Manual selected for thresholding method)</i><br>
-            Enter the value that will act as an absolute threshold for the images, a value from 0 to 1.
+            doc="""\
+*(Used only if Manual selected for thresholding method)*
+Enter the value that will act as an absolute threshold for the images, a
```
Comment (mcquin) [inline cellprofiler/modules/threshold.py]: Add a blank line above this line.

File: cellprofiler/modules/threshold.py [hunk 24]
```diff
@@ -352,48 +368,55 @@ def create_settings(self):
...
 
         self.thresholding_measurement = cellprofiler.setting.Measurement(
             "Select the measurement to threshold with",
             lambda: cellprofiler.measurement.IMAGE,
-            doc="""
-            <i>(Used only if Measurement is selected for thresholding method)</i><br>
-            Choose the image measurement that will act as an absolute threshold for the images.
+            doc="""\
+*(Used only if Measurement is selected for thresholding method)*
+Choose the image measurement that will act as an absolute threshold for
```
Comment (mcquin) [inline cellprofiler/modules/threshold.py]: Add a blank line above this line.

File: cellprofiler/modules/threshold.py [hunk 25]
```diff
@@ -498,7 +534,7 @@ def create_settings(self):
         self.adaptive_window_size = cellprofiler.setting.Integer(
             "Size of adaptive window",
             50,
-            doc="""
+            doc="""\
             Enter the window for the adaptive method. For example, you may want to use a multiple of the
```
Comment (mcquin) [inline cellprofiler/modules/threshold.py]: Shift text all the way to the left.

File: cellprofiler/modules/threshold.py [hunk 26]
```diff
@@ -1,15 +1,13 @@
 # coding=utf-8
 
 """
-Apply Threshold
+Threshold
```
Comment (bethac07) [inline cellprofiler/modules/threshold.py]: Can you remove some of the equal signs below to be equal in length to the word Threshold?

File: cellprofiler/modules/threshold.py [hunk 27]
```diff
@@ -498,9 +546,9 @@ def create_settings(self):
         self.adaptive_window_size = cellprofiler.setting.Integer(
             "Size of adaptive window",
             50,
-            doc="""
-            Enter the window for the adaptive method. For example, you may want to use a multiple of the
-            largest expected object size.
+            doc="""\
+Enter the window for the adaptive method. For example, you may want to use a multiple of the
```
Comment (bethac07) [inline cellprofiler/modules/threshold.py]: Please add `*(Used only if "{TS_ADAPTIVE}" is selected for thresholding scope)*` and a newline.

**Pull request merged**